src/_pytest/reports.py

```diff
@@ -74,6 +74,10 @@ def __init__(self, **kw: Any) -> None:
         def __getattr__(self, key: str) -> Any:
             ...
 
+        # Allow dynamic attributes to be set in tests without typing errors.
+        def __setattr__(self, key: str, value: Any) -> None:
+            ...
```

Comment:
there are couple of places in tests where I removed `# type: ignore[attr-defined]` - should I have kept them instead of adding `__setattr__` stub?

Reply:
I think, since we are probably going to make the `TestReport`/`CollectReport` types public at some point, it will be better to keep the type-ignores in the tests for now, to maintain our flexibility in terms of compatibility. If we find there's a good reason to set attributes after the fact instead of only through the `__init__`, we can reconsider then.

Reply:
reverted in 5d4284f